source/process/regex-library-draft.md

```diff
@@ -0,0 +1,229 @@
+
+</details>
+
+<details close>
+
+<summary>Footnotes</summary>
+
+* **footnote references: **for footnotes _not_ in `backmatter` use this find and replace to format footnote refs in each file. Adjust the find to match source file markup, if necessary, and edit the replace to ensure unique IDs. After replacing in BBEdit use _Markup > Update > Document_ to change `#FILENAME#` to document filename<br>F: `<p>(\d)\. (.*?)</p>`<br>R: `<div epub:type="footnote" id="\1">\n          <p><sup><a href="#FILENAME##backlink-\1">\1</a></sup>\&#160;<span class="note">\2</span></p>\n        </div>`
+* **footnote indicators: **for footnotes _not_ in `backmatter` use this find and replace to format footnote indicators in each file. Adjust the find to match source file markup, if necessary, and edit the replace to ensure unique IDs. After replacing in BBEdit use _Markup > Update > Document_ to change `#FILENAME#` to document filename<br>F: `<sup>(\d+)</sup>`<br>R: `<sup class="fn" id="backlink-intro-\1"><a epub:type="noteref" href="#FILENAME##intro-\1">[\1]</a></sup>`
+* **unique footnote reference id**: use filename to make footnote reference id unique<br>F: `<sup class="fn" id="note-backlink-(\d+)"><a epub:type="noteref" href="([^#]+)_([^#]*?).xhtml#note-(\d+)">\[(\d+)\]</a></sup>`<br>R: `<sup class="fn" id="note-backlink-\3-\1"><a epub:type="noteref" href="\2_\3.xhtml#note-\3-\4">[\5]</a></sup>`
+* **unique footnote indicator id**: use filename to make footnote id unique<br>F: `<div id="note-(\d+)" epub:type="footnote">\s*<p><sup><a href="([^#]+)_([^#]*?)\.xhtml#note-backlink-(\d+)">`<br>R: `<div id="note-\3-\1" epub:type="footnote"><p><sup><a href="\2_\3.xhtml#note-backlink-\3-\4">`
+* **remove Ibids: **make sure footnotes are formatted correctly according to the style guide and then use to replace Ibids<br>F: `(<p class="[^"]*"><sup>(\d+)</sup>(.*?<span class="i">.*?</span>).*?</p>\s*<p class="[^"]*"><sup>\d+</sup>)Ibid\.(,.*?)*</p>`<br>R: `\1\3\4</p>`
+
+</details>
+
+<details close>
+
+<summary>Index</summary>
+
+* **move pagebreaks up top**: find pagebreaks in a file and move them before the h1. (<mark>Run multiple times until there are no new finds</mark>)<br>F: `(<h1[^>]*>.*?</h1>(?msi)(.*?))(<span epub:type="pagebreak"[^>]*></span>)`<br>R: `\3\1`
+
+</details>
+
+<details close>
+
+<summary>Links</summary>
+
+* **add `target="_blank"` to links**: Add `target="_blank"` attribute to existing external links<br>F: `<a href="http([^"]+)"><br>R: <a href="http\1" target="_blank" rel="noopener">`<br>R: `<a href="http\1" target="_blank" rel="noopener">`
+  > Links to our own hosted videos should not have `target=_blank or rel="noopener"` attributes, but this RegEx will add them, so avoid using this RegEx unmodified on books with videos.
```

Comment:
You could put this paragraph in an `<aside class="caution">` to have it look like this:
![Screen Shot 2020-04-22 at 6 39 40 AM](https://user-images.githubusercontent.com/10226589/79988926-16108980-8464-11ea-8501-5c35fca19e03.png)
as easy as:
```
<aside class="caution">Links to our own hosted videos should not have `target="_blank"` or `rel="noopener"` attributes, but this RegEx will add them, so avoid using this RegEx unmodified on books with videos.</aside>
```


FYI, There are several different aside classes available in the theme:
"notice":
![Screen Shot 2020-04-22 at 6 30 06 AM](https://user-images.githubusercontent.com/10226589/79987889-d1d0b980-8462-11ea-99c6-7c10d06b8f0b.png)

"tip":
![Screen Shot 2020-04-22 at 6 32 11 AM](https://user-images.githubusercontent.com/10226589/79988055-0cd2ed00-8463-11ea-977a-5b00285f67d7.png)


"caution", (shown above)

and "warning":
![Screen Shot 2020-04-22 at 6 34 13 AM](https://user-images.githubusercontent.com/10226589/79988280-4e639800-8463-11ea-9544-a30192ddbf16.png)